### PR TITLE
CIDEVSTC-278 - Adds test to verify parameter functions can be added to coverage

### PR DESCRIPTION
**Pull request by lukecampbell**
The functionality to append parameters is there, I think there's an issue with it when it comes to complex coverages but as far as parameter functions go, it appears to work globally. This test proves that it works for Python Functions, there's another test already in the file that does it for numexpr functions in a complex coverage. I would say that it's done and verified at this point.

### CODE REVIEW

**Comment by lukecampbell** [comment]
@MauriceManning ready for review.
@mmeisinger for your information.

File: coverage_model/test/test_complex_coverage.py [hunk 1]
```diff
@@ -72,9 +72,40 @@ def _make_cov(root_dir, params, nt=10, data_dict=None, make_temporal=True):
...
+        dens_pmap = {'SP':'pracsal', 't':'seawater_temperature', 'p':'seawater_pressure', 'lat':'lat', 'lon':'lon'}
+        dens_expr = PythonFunction('density', owner, dens_func, dens_arglist, None, dens_pmap)
+        dens_ctxt = ParameterContext('density', param_type=ParameterFunctionType(dens_expr), variability=VariabilityEnum.TEMPORAL)
+        dens_ctxt.uom = 'kg m-3'
+
+        cov.append_parameter(dens_ctxt)
+
+        # Make sure it worked
+        np.testing.assert_array_equal(cov._range_value['density'][:], 
+                np.array([ 1024.98205566,  1019.4932251 ], dtype=np.float32))
```
Comment (lukecampbell) [inline coverage_model/test/test_complex_coverage.py:106]: Made a CTDBP-based coverage model without density defined. Then I added it dynamically and verified that the output is correct.

**Comment by MauriceManning** [comment]
:thumbsup: 
will wait on Michael to have a look before closing.

**Comment by mmeisinger** [comment]
looks good to me but no detailed insight

**Pull request merged**